### PR TITLE
GreenField: Create/Get/Delete Stores

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
I think a really cool feature in the API would be a "Clone" so that someone can automatically configure a template store and duplicate it for several customer. Maybe another PR though

**Comment by Kukks** [comment]
yeah another PR, I need to add update store to this too anyway

File: BTCPayServer/Controllers/GreenField/StoresController.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            }
+            await _storeRepository.RemoveStore(storeId, _userManager.GetUserId(User));
+            return Ok();
+        }
+        
+        [HttpPost("~/api/v1/stores")]
+        [Authorize(Policy = Policies.CanModifyStoreSettings, AuthenticationSchemes = AuthenticationSchemes.Greenfield)]
+        public async Task<ActionResult<Client.Models.StoreData>> CreateStore(CreateStoreRequest request)
+        {
+            if (request is null)
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/StoresController.cs:70]: `request?.Name is null`

File: BTCPayServer/Controllers/GreenField/StoresController.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        private readonly StoreRepository _storeRepository;
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public StoresController(StoreRepository storeRepository, UserManager<ApplicationUser> userManager)
+        {
+            _storeRepository = storeRepository;
+            _userManager = userManager;
+        }
+        [Authorize(Policy = Policies.CanViewStoreSettings, AuthenticationSchemes = AuthenticationSchemes.Greenfield)]
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/GreenField/StoresController.cs:27]: `CanViewStoreSettings` is a scoped permission. Should use a separate permission.

**Pull request merged**